roxctl/central/m2m/exchange/exchange.go

```diff
@@ -0,0 +1,121 @@
+package exchange
+
+import (
+	"bytes"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common/auth"
+	"github.com/stackrox/rox/roxctl/common/config"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+	"github.com/stackrox/rox/roxctl/common/util"
+)
+
+// Command to exchange an OIDC token for a short-lived access token.
+func Command(cliEnvironment environment.Environment) *cobra.Command {
+	exchangeCmd := exchangeCommand{env: cliEnvironment}
+
+	cmd := &cobra.Command{
+		Use:   "exchange",
+		Short: "Exchanges a OIDC token for a short-lived access token",
+		Long: `Obtain a short-lived access token from Central by exchanging an OIDC token.
+This works by configuring a machine-to-machine access configuration within Central beforehand.
+Based on the OIDC token's issuer, a short-lived access token will be exchanged.`,
```

Comment:
Do we have e2e for that?

Reply:
Currently not, no. This requires besides the command to be executed also a) existing config within Central and b) a valid OIDC provider wtihin CI from which you obtain an ID token to pass to Central.

I'll look into that in the future how easy it'll be with our current testing setup, but sadly it does seem to be some significant effort for the time being (at the least, if we run this in prow).

Reply:
I think it would be good if the documentation of the command made it clear that the obtained token is not exposed in the output, but stored in the `roxctl` configuration file.